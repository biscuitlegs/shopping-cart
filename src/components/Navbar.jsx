import React, { useEffect } from 'react'
import { Link, useLocation } from 'react-router-dom'

const Navigation = () => {
  const currentPath = useLocation().pathname

  const setLinksInactive = (links) => {
    links.forEach((link) => {
      link.classList.remove('active')
      link.removeAttribute('aria-current')
    })
  }

  const setLinkActive = (link) => {
    link.classList.add('active')
    link.setAttribute('aria-current', 'page')
  }

  useEffect(() => {
    const homeLink = document.querySelector('a[href="/"].nav-link')
    const shopLink = document.querySelector('a[href="/shop"].nav-link')
    setLinksInactive([homeLink, shopLink])
    switch (true) {
      case currentPath === '/':
        setLinkActive(homeLink)
        break
      case currentPath === '/shop':
        setLinkActive(shopLink)
        break
      default:
        break
    }
  })

  return (
    <nav className="navbar navbar-expand-lg bg-light">
      <div className="container-fluid">
        <Link className="navbar-brand" to="/">
          React Shop
        </Link>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav me-auto mb-2 mb-lg-0">
            <li className="nav-item">
              <Link className="nav-link active" aria-current="page" to="/">
                Home
              </Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link" to="/shop">
                Shop
              </Link>
            </li>
          </ul>
          <button
            type="button"
            className="btn btn-primary"
            data-bs-toggle="modal"
            data-bs-target="#exampleModal"
          >
            <span className="me-2">Basket</span>
            <i className="bi bi-bag"></i>
          </button>
        </div>
      </div>
    </nav>
  )
}

export default Navigation
