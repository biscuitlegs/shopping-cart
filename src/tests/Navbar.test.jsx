import React from 'react'
import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Navbar from '../components/Navbar.jsx'

// We test the actual navigation functionality in 'App.test.jsx'
// because the routes are in the App component rather than here

test('shows a navbar on the screen', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )

  const nav = screen.getByRole('navigation')

  expect(nav).toBeInTheDocument()
})

test('shows a brand link in the navbar', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )

  const brandLink = screen.getByRole('link', { name: 'React Shop' })

  expect(brandLink).toBeInTheDocument()
})

test('shows a home link in the navbar', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )

  const homeLink = screen.getByRole('link', { name: 'Home' })

  expect(homeLink).toBeInTheDocument()
})

test('shows a shop link in the navbar', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )

  const shopLink = screen.getByRole('link', { name: 'Shop' })

  expect(shopLink).toBeInTheDocument()
})

test('shows a basket button in the navbar', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )

  const button = screen.getByRole('button', { name: /Basket/ })

  expect(button).toBeInTheDocument()
})
