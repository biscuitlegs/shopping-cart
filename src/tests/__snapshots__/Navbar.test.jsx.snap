// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar correctly 1`] = `
<nav
  className="navbar navbar-expand-lg bg-light"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      React Shop
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/shop"
            onClick={[Function]}
          >
            Shop
          </a>
        </li>
      </ul>
      <button
        className="btn btn-primary"
        data-bs-target="#exampleModal"
        data-bs-toggle="modal"
        type="button"
      >
        <span
          className="me-2"
        >
          Basket
        </span>
        <i
          className="bi bi-bag"
        />
      </button>
    </div>
  </div>
</nav>
`;
